### DISCUSSION TITLE
Welcome to hacktribe Discussions!

### DISCUSSION THREAD

**Discussion by bangcorrupt** [bangcorrupt/hacktribe#22] [Announcements]
## 👋 Welcome!
  This is a place to ask questions, get help and discuss anything about hacktribe.  

Please see the [FAQ](https://github.com/bangcorrupt/hacktribe/wiki/FAQ), read the [Wiki](https://github.com/bangcorrupt/hacktribe/wiki) and [search discussions](https://docs.github.com/en/search-github/searching-on-github/searching-discussions) before posting a new thread, thank you.

**Comment by ptowns** [comment]
Thank you so much for this! It's amazing. I'm curious of the inner workings of the synth engine? Also have you found anything about the master compressor or master channel? I always hear compression?  If this is coverd somewhere please direct me to further reading! Are you on Instagram? I'd love to pick your brain.

**Reply by bangcorrupt** [reply]
I haven't looked into the sound engine very far yet, only how the presets work for oscillators and FX.

**Comment by ShoveD01** [comment]
Hi good work, I used the get-filehash command to display the actual checksums, and they are very different , but then again they would be, wouldn't they, given the code has been modified?

**Reply by bangcorrupt** [reply]
Before patching the firmware, sha256 hash of `SYSTEM.VSB` should match the hash in `hash/SYSTEM.VSB.sha`.  This proves you have the correct version of the factory firmware. 

After patching the firmware, sha256 hash of `hacked-SYSTEM.VSB` should match the hash in `hash/hacked-SYSTEM.VSB.sha`.  This proves you have the same modified firmware I have tested.

**Comment by ShoveD01** [comment]
Ah , I get it now . Thanks for that.

**Comment by ShoveD01** [comment]
That’s weird it doesn’t , I am using the get-filehash command that displays the actual checksum, and they don’t match.

**Comment by ShoveD01** [comment]
They are different file sizes too

**Comment by bangcorrupt** [comment]
Can you paste the terminal output?

**Comment by ShoveD01** [comment]
Attached picture
The two errors were caused by missing the .sha extension

**Reply by bangcorrupt** [reply]
If you attached an image I can't see it. I don't think attachments will work through the github notification emails. Can you either copy and paste the text, or post a link to an image hosted elsewhere?

**Comment by ShoveD01** [comment]


**Reply by bangcorrupt** [reply]
Ok, I see what happened.  Your first command `get-filehash hacked-SYSTEM.VSB` is correct and shows the firmware has been modified correctly.  This computes a hash of the modified firmware file.  The hash displayed matches the hash found in `hash/hacked-SYSTEM.VSB.sha`.  

In the next command `get-filehash hash\hacked-SYSTEM.VSB.sha` you are computing a hash of the hash file.  This is not necessary.

**Comment by ShoveD01** [comment]
How do I see the hash of the hashed file?
Also I have compared the hash of the newly downloaded firmware and the same file which was download a couple of years ago , and they are different.

**Comment by ShoveD01** [comment]
Ah got it , notepad.    Thanks for your help.

**Comment by ShoveD01** [comment]
Once again thanks for the work you've done and your help.
On the old hack samples could be loaded in slots >19 ( but avoiding the chopped samples) however the sample names did not then tally, is this still the case?

**Reply by bangcorrupt** [reply]
No problem at all. To avoid errors, don't load any samples to slots <501. This may change in the future but it's low priority.

**Reply by ShoveD01** [reply]
I haven’t installed the hack yet , are the original sample names still visible?

**Reply by bangcorrupt** [reply]
No, the original samples are not loaded from the flash.  If you have a factory sampler you can export the factory samples and load them into the user slots.

**Comment by ShoveD01** [comment]
I’m sure I’ve got the original samples somewhere

**Comment by ShoveD01** [comment]
Yes I’ve got them as a .all.  From hippy at 19.  To shaker at 335

**Reply by bangcorrupt** [reply]
If you edit the .all so the first sample is in slot 501 they will load correctly.

**Reply by ShoveD01** [reply]
Thanks , what are the sysex strings that currently work with the e2?

**Reply by bangcorrupt** [reply]
See #37.

**Comment by Obuzus** [comment]
Is posible to help with more exact information about the debrick procedure?

**Reply by bangcorrupt** [reply]
Answered in #66.

**Comment by ShoveD01** [comment]
Just noticed the hacktribe is sampling at 48k, is that the normal sample rate for the unhacked sampler, or could that be the reason for the higher pitched sample bug?

**Reply by bangcorrupt** [reply]
I think 48k is standard.  Sample pitch bug is fixed in the latest commit.  Which also appears to have fixed #4.
